ingestion/src/metadata/orm_profiler/profiler/runner.py

```diff
@@ -55,25 +59,76 @@ def _build_query(self, *entities, **kwargs) -> Query:
         return self._session.query(*entities, **kwargs)
 
     def _select_from_sample(self, *entities, **kwargs):
-        return self._build_query(*entities, **kwargs).select_from(self._sample)
+        """Run select statement against sample data"""
+        if kwargs.get("query_filter_"):
+            query_filter = kwargs.pop("query_filter_")
+            filter_ = build_query_filter(**query_filter)
+        else:
+            filter_ = None
+
+        query = self._build_query(*entities, **kwargs).select_from(self._sample)
+
+        if filter_ is not None:
+            return query.filter(filter_)
+
+        return query
 
     def _select_from_user_query(self, *entities, **kwargs):
+        """Run select statement against user defined query"""
+        if kwargs.get("query_filter_"):
+            query_filter = kwargs.pop("query_filter_")
+            filter_ = build_query_filter(**query_filter)
+        else:
+            filter_ = None
+
         user_query = self._session.query(self.table).from_statement(
             text(f"{self._profile_sample_query}")
         )
-        return self._build_query(*entities, **kwargs).select_from(user_query)
+
+        query = self._build_query(*entities, **kwargs).select_from(user_query)
+
+        if filter_ is not None:
+            return query.filter(filter_)
+
+        return query
 
     @partition_filter_handler()
     def select_first_from_table(self, *entities, **kwargs):
+        """Select first row from the table"""
+        if kwargs.get("query_filter_"):
```

Comment:
I think these lines 

```python
if kwargs.get("query_filter_"):
    query_filter = kwargs.pop("query_filter_")
    filter_ = build_query_filter(**query_filter)
else:
    filter_ = None
```

are repeated a few times. Maybe we could have a function wrapping this